File: src/main/java/com/geo/analytics/web/dto/CreateJobRequest.java
package com.geo.analytics.web.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.geo.analytics.domain.enums.CompetitorExtractionMode;
import com.geo.analytics.domain.support.TextWhitespaceNormalizer;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import java.util.UUID;

public record CreateJobRequest(
    @NotBlank(message = "brandName must not be blank")
    @Size(max = 255, message = "brandName must not exceed 255 characters")
    String brandName,
    @NotBlank(message = "targetUrl must not be blank")
    @Size(max = 2048, message = "targetUrl must not exceed 2048 characters")
    String targetUrl,
    @Size(max = 16000)
    String businessSummary,
    @Size(max = 16000)
    String targetAudience,
    @Size(max = 16000)
    String focusPoints,
    @JsonProperty("industryType") CompetitorExtractionMode competitorExtractionMode,
    @JsonProperty("idempotency_key") UUID idempotencyKey
) {
    public CreateJobRequest {
        brandName = TextWhitespaceNormalizer.normalize(brandName);
        targetUrl = TextWhitespaceNormalizer.normalize(targetUrl);
        businessSummary = optionalText(businessSummary);
        targetAudience = optionalText(targetAudience);
        focusPoints = optionalText(focusPoints);
        if (competitorExtractionMode == null) {
            competitorExtractionMode = CompetitorExtractionMode.LOCAL_STORE;
        }
    }

    private static String optionalText(String value) {
        if (value == null) {
            return null;
        }
        String s = value.strip();
        return s.isEmpty() ? null : s;
    }
}

File: frontend/src/types/createJobRequest.ts
export type CompetitorExtractionMode = "LOCAL_STORE" | "CORPORATE_SERVICE" | "ONLINE_SERVICE";

export type CreateJobRequestPayload = {
  brandName: string;
  targetUrl: string;
  industryType?: CompetitorExtractionMode;
  businessSummary?: string;
  targetAudience?: string;
  focusPoints?: string;
  files?: File[];
};

function optionalTrim(value: string | undefined): string | undefined {
  if (value === undefined) {
    return undefined;
  }
  const t = value.trim();
  return t === "" ? undefined : t;
}

export function buildCreateJobBody(input: CreateJobRequestPayload): Record<string, unknown> {
  const out: Record<string, unknown> = {
    brandName: input.brandName.trim(),
    targetUrl: input.targetUrl.trim(),
    industryType: input.industryType ?? "LOCAL_STORE",
  };
  const businessSummary = optionalTrim(input.businessSummary);
  const targetAudience = optionalTrim(input.targetAudience);
  const focusPoints = optionalTrim(input.focusPoints);
  if (businessSummary !== undefined) {
    out.businessSummary = businessSummary;
  }
  if (targetAudience !== undefined) {
    out.targetAudience = targetAudience;
  }
  if (focusPoints !== undefined) {
    out.focusPoints = focusPoints;
  }
  return out;
}
